src/pulselistener/pulselistener/lib/bus.py

```diff
@@ -12,30 +12,22 @@ class MessageBus(object):
     '''
     Communication bus between processes
     '''
-    def __init__(self, max_messages=None):
+    def __init__(self):
         self.queues = {}
-        self.max_messages = max_messages
-        self.nb_messages = 0
 
-    def add_queue(self, name, mp=False):
+    def add_queue(self, name, mp=False, maxsize=-1):
         '''
         Create a new queue on the message bus
         * asyncio by default
         * multiprocessing when mp=True
+        By default, there are no size limit enforced (maxsize=-1)
         '''
         assert name not in self.queues, 'Queue {} already setup'.format(name)
+        assert isinstance(maxsize, int)
         if mp:
-            self.queues[name] = multiprocessing.Queue()
+            self.queues[name] = multiprocessing.Queue(maxsize=maxsize)
```

Comment:
In the multiprocessing docs, it's not specified what happens when maxsize < 0, but I guess it's the same behavior as asyncio.Queue. Maybe you can confirm with a simple script to put a lot of things into a multiprocessing.Queue and see that it never blocks.

Reply:
It says [here](https://docs.python.org/3.5/library/multiprocessing.html#multiprocessing.Queue):
> Queue implements all the methods of queue.Queue except for task_done() and join().

So i took that as the same behaviour was implemented.

I'll add the test though, to be sure

Reply:
I meant just testing it locally on your machine to confirm, but we don't need to add a test here

Reply:
Oups.

Reply:
Well, it won't hurt ;)